app/src/main/java/com/tomtom/sdk/examples/usecase/MapModeSelectionActivity.kt

```diff
@@ -0,0 +1,145 @@
+    private var onlineButton: MaterialCardView? = null
+    private var offlineButton: MaterialCardView? = null
+    private var hybridButton: MaterialCardView? = null
+    private var mapMode = MapMode.ONLINE
+
+    companion object {
+        private const val STROKE_WIDTH = 4
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_map_mode_selection)
+        val toolbar = findViewById<View>(R.id.toolbar) as? Toolbar
+        setSupportActionBar(toolbar);
+
+        supportActionBar?.setDisplayHomeAsUpEnabled(true);
+
+        val onlineButtonId = R.id.online_button
+        val hybridButtonId = R.id.hybrid_button
+        val offlineButtonId = R.id.offline_button
+
+        onlineButton = findViewById(onlineButtonId)
+        hybridButton = findViewById(hybridButtonId)
+        offlineButton = findViewById(offlineButtonId)
+
+        onlineButton?.isSelected = true
+
+        onlineButton?.setOnClickListener {
+            selectButton(onlineButtonId)
+            mapMode = MapMode.ONLINE
+        }
+
+        /*
+        hybridButton?.setOnClickListener {
+            selectButton(hybridButtonId)
+            mapMode = MapMode.HYBRID
+
+            // In order to activate, please remove the comment lines,
+            // and remove the darker gray tint of ImageView in the xml file
+        }
+
+        offlineButton?.setOnClickListener {
+            selectButton(offlineButtonId)
+            mapMode = MapMode.OFFLINE
+
+            // In order to activate, please remove the comment lines,
+            // and remove the darker gray tint of ImageView in the xml file
+        }
+        */
+
+
+        findViewById<Button>(R.id.bt_proceed).setOnClickListener {
+            proceed()
+        }
+    }
+
+    private fun proceed() {
+        when (mapMode) {
+            MapMode.ONLINE -> {
+                val myIntent = Intent(this, BasicNavigationActivity::class.java)
+                this.startActivity(myIntent)
+            }
+            MapMode.HYBRID -> {
+                TODO()
+            }
+            MapMode.OFFLINE -> {
+                TODO()
+            }
+        }
+    }
+
+    private fun selectButton(selectedButtonId: Int) {
+        onlineButton?.strokeWidth = 0
+        hybridButton?.strokeWidth = 0
+        offlineButton?.strokeWidth = 0
+
+        val density = resources.displayMetrics.density
+        val strokeWidthPixels = (STROKE_WIDTH * density).toInt()
+
+        findViewById<MaterialCardView>(selectedButtonId).strokeWidth = strokeWidthPixels
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+        menuInflater.inflate(R.menu.info_button, menu);
+
+        val infoMenuItem = menu?.findItem(R.id.action_info)
+
+        val iconTint = ContextCompat.getColor(this, R.color.active_color)
+        val icon = infoMenuItem?.icon
+        icon?.setColorFilter(iconTint, PorterDuff.Mode.SRC_IN)
```

Comment:
Why it is not specified in the XML?

Reply:
It was actually specified in XML, but i was playing with the colors, so this is just unnecessary code here. deleted it now. thanks